lightning/src/ln/channel.rs

```diff
@@ -3460,76 +3645,105 @@ impl<Signer: Sign> Channel<Signer> {
 			},
 		};
 
-		let closing_tx_max_weight = self.get_closing_transaction_weight(
-			if let Some(oup) = closing_tx.output.get(0) { Some(&oup.script_pubkey) } else { None },
-			if let Some(oup) = closing_tx.output.get(1) { Some(&oup.script_pubkey) } else { None });
-		if let Some((_, last_fee, sig)) = self.last_sent_closing_fee {
+		assert!(self.shutdown_scriptpubkey.is_some());
+		if let Some((last_fee, sig)) = self.last_sent_closing_fee {
 			if last_fee == msg.fee_satoshis {
 				self.build_signed_closing_transaction(&mut closing_tx, &msg.signature, &sig);
-				assert!(closing_tx.get_weight() as u64 <= closing_tx_max_weight);
```

Comment:
Do we wanna migrate these assertions?

Reply:
Same answer as here, I think https://github.com/rust-bitcoin/rust-lightning/pull/1011#discussion_r688701553